src/04-test-class/index.test.ts

```diff
@@ -1,44 +1,64 @@
 // Uncomment the code below and write your tests
-// import { getBankAccount } from '.';
+import { getBankAccount, InsufficientFundsError, SynchronizationFailedError, TransferFailedError } from '.';
 
 describe('BankAccount', () => {
   test('should create account with initial balance', () => {
-    // Write your test here
+    const bankAccount = getBankAccount(300);
+    expect(bankAccount.getBalance()).toBe(300);
   });
 
   test('should throw InsufficientFundsError error when withdrawing more than balance', () => {
-    // Write your test here
+    const bankAccount = getBankAccount(300);
+    expect(() => bankAccount.withdraw(500)).toThrow(InsufficientFundsError);
   });
 
   test('should throw error when transferring more than balance', () => {
-    // Write your test here
+    const bankAccount = getBankAccount(300);
+    expect(() => bankAccount.withdraw(500)).toThrow(InsufficientFundsError);
   });
 
   test('should throw error when transferring to the same account', () => {
-    // Write your test here
+    const bankAccount = getBankAccount(300);
+    expect(() => bankAccount.transfer(500, bankAccount)).toThrow(TransferFailedError);
   });
 
   test('should deposit money', () => {
-    // Write your test here
+    const bankAccount = getBankAccount(300);
+    expect(bankAccount.deposit(200).getBalance()).toBe(500);
   });
 
   test('should withdraw money', () => {
-    // Write your test here
+    const bankAccount = getBankAccount(300);
+    expect(bankAccount.withdraw(200).getBalance()).toBe(100);
   });
 
   test('should transfer money', () => {
-    // Write your test here
+    const  bankAccount1 = getBankAccount(300);
+    const  bankAccount2 = getBankAccount(100);
+    expect(bankAccount1.transfer(200, bankAccount2).getBalance()).toBe(100);
+    expect(bankAccount2.getBalance()).toBe(300);
   });
 
   test('fetchBalance should return number in case if request did not failed', async () => {
-    // Write your tests here
+    const bankAccount = getBankAccount(300);
+    const fetchBalance = await bankAccount.fetchBalance();
```

Comment:
'case request didnt failed' - you implementation dont know about request fail. It runs only if fetchBalance returns !null